lib/persist-state.c

```diff
@@ -267,6 +314,12 @@ _alloc_value(PersistState *self, guint32 orig_size, gboolean in_use, guint8 vers
   persist_state_unmap_entry(self, self->current_ofs);
 
   self->current_ofs += size + sizeof(PersistValueHeader);
+
+  if (!_check_watermark(self) && !_grow_store(self, self->current_size + PERSIST_FILE_INITIAL_SIZE))
+    {
+      main_loop_exit();
+    }
+
   return result;
```

Comment:
@bazsi, sorry, but I think that main_loop_exit is a callback, because this is an ivykis event, where mainloop registers, do I misunderstand something?